CHANGELOG.md

```diff
@@ -75,7 +75,7 @@
 
   - General data model
     - `DataPayload` no longer needs a lifetime (#1297, #1279)
-    - Re-write ResourceKey (#1511)
+    - Re-write DataKey (#1511)
```

Comment:
Nit: Don't backport name changes to docs and changelog that are temporal

Reply:
We have actually done this in other renamings as well (Format -> Formatter). I think it makes sense, these are clean renames and the change log will be harder to interpret if the names in it don't exist anymore.